core-bundle/src/Resources/contao/library/Contao/Database/Statement.php

```diff
@@ -253,7 +253,13 @@ public function query($strQuery='', array $arrParams = array(), array $arrTypes
 		$arrParams = array_map(
 			static function ($varParam)
 			{
-				if (\is_string($varParam) || \is_bool($varParam) || \is_float($varParam) || \is_int($varParam) || $varParam === null)
+				// Cast boolean to integer, otherwise PDO will convert "false" to an empty string (see https://bugs.php.net/bug.php?id=57157)
```

Comment:
not sure I like this. We should include the type information of columns, then Doctrine would automatically convert the values according to the database abstraction (not all platforms use integer as boolean). Not sure if we can do that in all places though, maybe we only cast if the column is not given in `$arrTypes`?

Reply:
> Not sure if we can do that in all places though

We can in fact not do it with `Contao\Database\Statement::set()` currently. Also it would be cumbersome to do that in any case.

However yes, we could at least check for `empty($arrTypes)`.

Reply:
now we can 😎 https://github.com/contao/contao/pull/4839
Yes it might be cumbersome, but that's just how Doctrine works. If you use boolean fields in DBAL, you must include type information for it to work. Maybe our model abstraction could automatically include it. Also, I'm fine with still casting if there is no type 🙃. Just be aware that params and types are NOT keyed, they are by order.